python_modules/dagster/dagster/_core/definitions/run_request.py

```diff
@@ -352,6 +356,48 @@ def __new__(
         )
 
 
+@whitelist_for_serdes
+@experimental
+class BackfillRequest(
+    NamedTuple(
+        "_BackfillRequest",
+        [
+            ("asset_partitions", "AssetGraphSubset"),
+            ("tags", PublicAttr[Optional[Mapping[str, str]]]),
+        ],
+    )
+):
+    @property
+    def asset_keys(self) -> AbstractSet[AssetKey]:
+        return self.asset_partitions.asset_keys
+
+    @classmethod
+    def from_asset_selection_and_partitions(
+        cls,
+        asset_selection: Sequence[AssetKey],
+        partition_keys: Sequence[str],
+        asset_graph: "AssetGraph",
+        tags: Optional[Mapping[str, str]] = None,
+    ):
+        from dagster._core.definitions.asset_graph_subset import AssetGraphSubset
+
+        partitions_subsets_by_asset_key: Dict[AssetKey, PartitionsSubset] = {}
+        non_partitioned_asset_keys: Set[AssetKey] = set()
+
+        for asset_key in asset_selection:
+            partitions_def = asset_graph.get_partitions_def(asset_key)
+            if partitions_def:
+                partitions_subsets_by_asset_key[
+                    asset_key
+                ] = partitions_def.subset_with_partition_keys(partition_keys)
```

Comment:
This method doesn't enforce validity (since partitions subsets can contain partitions that have since been deleted). 

I think we should add some checks here to prevent empty subsets / subsets with nonexistent partitions.

Reply:
Off the top of your head, are there other places in the codebase where we enforce validity that I could copy from?

Reply:
https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/execution/asset_backfill.py#L803-L807

Here's an example, though this might be slow if many time partitions are targeted... 

It might be better to allow construction of invalid subsets here (since most sensors have the 30s time limit), and enforce validity within the daemon

Reply:
do you have a sense of how long this typically takes in the daemon? if we're incurring this on every backfill tick in the daemon, that also seems unideal?

Reply:
but this makes sense. I'm going to add a test that encodes that it's OK to create backfills with nonexistent partitions.

Reply:
It would be the amount of time it takes to fetch all partition keys + do the AND operation against the partition keys in the subset. When we did testing last year, this took 10s for a time partitions def with 100K partitions

Reply:
And I mean doing the check in `_handle_backfill_requests`, not in the backfill daemon. So we only need to do the check once per backfill request, which also enforces that created backfills are valid.

Reply:
Got it - one consequence here is we could end up in a situation where we launch some runs and instigate some backfills based on the sensor result, but then raise an error because of an invalid backfill request. The tick status would then be ERROR, even though we had successfully done some work on behalf of the backfill.

Curious if @gibsondan has thoughts on this situation and whether it's problematic or not.